src/packages/ufm/components/ufm-element-base.ts

```diff
@@ -36,6 +36,10 @@ export abstract class UmbUfmElementBase extends UmbLitElement {
 		});
 	}
 
+	override createRenderRoot() {
+		return this;
+	}
```

Comment:
This is niggling at me, as inheriting from the `UmbUfmElementBase` class isn't a requirement for a UFM extension/component, so 3rd-party extensions may not implement this and lead to missing text/content in the rendered label.

I'm wondering if we could traverse the Shadow DOM to gather all the text content from the children/descendants?

e.g.

```ts
// snipped from ufm-virtual-render.controller.ts

override toString(): string {
	return this.#getTextFromDescendants(this.#element);
}

#getTextFromDescendants(element?: Element | null): string {
	if (!element) return '';

	const items: Array<string> = [];

	items.push(element.shadowRoot?.textContent ?? element.textContent ?? '');

	if (element.shadowRoot !== null) {
		Array.from(element.shadowRoot.children).forEach((element) => {
			items.push(this.#getTextFromDescendants(element));
		});
	}

	if (element.children !== null) {
		Array.from(element.children).forEach((element) => {
			items.push(this.#getTextFromDescendants(element));
		});
	}

	return items.filter((x) => x).join(' ');
}
```

There may be a more succinct way in code to do this, but you get the general idea.